lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -0,0 +1,413 @@
+        """Test function returns a list containing the original cube list if
+        the accumulation_period is not set."""
+
+        time_interval = 60
+        plugin = Accumulation()
+        result = plugin.get_period_sets(time_interval, self.cubes)
+        self.assertSequenceEqual(result, [self.cubes])
+
+    @ManageWarnings(record=True)
+    def test_raises_warning_for_unused_cubes(self, warning_list=None):
+        """Test function raises a warning when there are insufficient cubes to
+        complete the last period."""
+
+        time_interval = 60
+        warning_msg = (
+            "The provided cubes result in a partial period given the specified"
+            " accumulation_period, i.e. the number of cubes is insufficient to"
+            " give a set of complete periods. Only complete periods will be"
+            " returned.")
+
+        expected = [self.cubes[0:4], self.cubes[3:7], self.cubes[6:10]]
+
+        plugin = Accumulation(accumulation_period=180)
+        result = plugin.get_period_sets(time_interval, self.cubes)
+
+        for index, sublist in enumerate(result):
+            self.assertSequenceEqual(sublist, expected[index])
+        self.assertTrue(any(item.category == UserWarning
+                            for item in warning_list))
+        self.assertTrue(any(warning_msg in str(item)
+                            for item in warning_list))
+
+    def test_raises_exception_for_impossible_aggregation(self):
+        """Test function raises an exception when attempting to create an
+        accumulation_period that cannot be created from the input cubes."""
+
+        time_interval = 61
+        plugin = Accumulation(accumulation_period=120)
+        msg = "The specified accumulation period "
+
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin.get_period_sets(time_interval, self.cubes)
+
+
+class Test_process(rate_cube_set_up):
+    """Tests the process method results in the expected outputs."""
+
+    def test_returns_cubelist(self):
+        """Test function returns a cubelist containing one less entry than the
+        input cube list."""
+
+        result = Accumulation(accumulation_period=60).process(self.cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        self.assertEqual(len(self.cubes) - 1, len(result))
+
+    def test_returns_masked_cubes(self):
+        """Test function returns a list of masked cubes for masked input
+        data."""
+
+        result = Accumulation().process(self.cubes)
+        self.assertIsInstance(result[0].data, np.ma.MaskedArray)
+
+    def test_default_output_units(self):
+        """Test the function returns accumulations in the default units if no
+        units are explicitly set, where the default is metres."""
+
+        # Multiply the rates in mm/s by 60 to get accumulation over 1 minute
+        # and divide by 1000 to get into metres.
+        expected = self.cubes[0].copy(data=self.cubes[0].data * 60 / 1000)
```

Comment:
As above: I think the setup needs modifying with different rates at different times, to show that the plugin correctly identifies and weights the data.